src/liballoc/lib.rs

```diff
@@ -69,7 +69,10 @@
        issue_tracker_base_url = "https://github.com/rust-lang/rust/issues/",
        test(no_crate_inject, attr(allow(unused_variables), deny(warnings))))]
 #![no_std]
-#![needs_allocator]
+
+// Don't inject an allocator in stage0 since we explicitly use alloc_system, jemalloc
+// being possibly unavailable or disallowed
+#![cfg_attr(not(stage0), needs_allocator)]
```

Comment:
I think the build should be able to complete even if this attribute is present (e.g. this crate always needs an allocator). Did you run into an error if it was present in stage0?